### DISCUSSION TITLE
Python Dependencies

### DISCUSSION THREAD

**Discussion by ghost** [Tecnativa/doodba#576] [Doodba - The template]
How do I add python dependencies? Adding them to xxxxxx/odoo/custom/dependencies/pip.txt didn't work.

**Comment by pedrobaeza** [comment]
That's the way to do it. Check if you have rebuilt the image after adding the dependency.

**Comment by ap-wtioit** [comment]
do you have the output of the build? (when changing dependencies you need to rebuild the image)

edit: also if you can share the contents of your pip.txt it would help. (some options used in pip.txt can break the `pip install` command (e.g. using override args (like --no-deps) on one line propagates to the next lines))

edit2: what you are looking for is a line like:
```
#11 5.601 doodba INFO: Executing: ['pip', 'install', '--no-cache-dir', '-r', '/opt/odoo/custom/dependencies/pip.txt']
```
and possible following errors

**Comment by joemar25** [comment]
you may just navigate to that folder and use cmd or terminal to use this command:

1. Go to your folder xxxxxx/odoo/custom/dependencies/
2.  Open CMD or Terminal
3. Use command: ```python  pip freeze > pip.txt ```

![image](https://github.com/Tecnativa/doodba/assets/80235976/c20a5642-6a21-4a3f-abb5-13c3070b5965)

![image](https://github.com/Tecnativa/doodba/assets/80235976/d70451c4-b32a-489e-ac00-aa7f0360f4dd)

If you want to add a new dependency to the pip.txt or requirements.txt file, you can do so by opening the file in a text editor, adding the new dependency (and optionally the version number) on a new line, and saving the file. For example, to add the requests library version 2.25.1, you would add a line that says requests==2.25.1.

![image](https://github.com/Tecnativa/doodba/assets/80235976/117e9b32-296e-4801-bde7-d3434bc5240f)

**Comment by ydccgfc** [comment]
do you have the output of the build? (when changing dependencies you need to rebuild the image)

edit: also if you can share the contents of your pip.txt it would help. (some options used in pip.txt can break the pip install command (e.g. using override args (like --no-deps) on one line propagates to the next lines))

edit2: what you are looking for is a line like:

#11 5.601 doodba INFO: Executing: ['pip', 'install', '--no-cache-dir', '-r', '/opt/odoo/custom/dependencies/pip.txt']
and possible following errors
